harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py

```diff
@@ -519,17 +519,10 @@ def on_trial_closed(
         if self.trial_tracker.should_stop:
             self._state_print_checks(searcher_state)
             # Shutdown if `should_stop` is True, and this was the last running trial.
-            completed_trials = searcher_state.trials_closed | searcher_state.failures
-            running_trials = searcher_state.trials_created - completed_trials
+            running_trials = searcher_state.trials_created - searcher_state.trials_closed
             new_ops_list = [searcher.Shutdown()] if not running_trials else []
-
         else:
-            metric = self.trial_tracker[request_id].metric
-            new_ops_list = self.get_new_searcher_ops_list(
```

Comment:
Why are we not doing this any more? Not sure I understand the context of this change

Reply:
This fixes a bug I introduced earlier when I was confused about what operations occur when.

The moral of the story is that successful trials will trigger `on_validation_completed`, failed trials will trigger `on_trial_exited_early`, and (as I learned) `on_trial_closed` will be triggered:
* Whenever an explicit `Close` operation is posted via `SearchRunner` code
* Whenever a Trial fails, irrespective of what you do in your `SearchRunner` code (as I learned and was surprised by)

So, basically if you want to do different stuff for failed and successful trials it is cleanest to just move all of the trial creation logic to `on_validation_completed` and `on_trial_exited_early`, and not put any creation logic in `on_trial_closed`.  I had meant to delete the creation code in `on_trial_closed` in a previous PR, but it looks like I forgot, so I fixed that here.